zinc/src/sbt-test/source-dependencies/sealed/test

```diff
@@ -5,5 +5,4 @@ $ copy-file changes/A.scala A.scala
 
 # D.scala needs recompiling because the pattern match in D
 # is no longer exhaustive, which emits a warning
-> checkRecompilations 1 A B C E
-> checkRecompilations 2 D
+-> compile
```

Comment:
> Why has this been changed?

Because what I expect as output is failure of compilation, as it was tested in sbt/sbt, and it's more direct result of what we've tested using this test. The comment right above says

```
 # D.scala needs recompiling because the pattern match in D
 # is no longer exhaustive, which emits a warning
```

See also `source-dependencies/check-recompilation` test for a more normal usage. `checkRecompilations` looks into the Analysis so see what step each classes were recompiled. The expected output we want is that the final compilation to show warning when the pattern match is no longer exhaustive. We are testing the behavior using `-Xfatal-warnings` flag.

```
> checkRecompilations 1 A B C E
> checkRecompilations 2 D
```

says, in 1st compilation step it compiled A, B, C, and E. And in the 2nd step, it compiled D. This does not tell us about the warning. In other words, this is a _weaker_ test than what we had in sbt 0.13.

Now thanks to the `-Xfatal-warnings` flag, it fails to compile. So I could potentially put `-> checkRecompilations 1 A B C E` (with minus sign), but that value would be nonsensical.

Reply:
The issue here is that `checkRecompilations` runs both compile and checks the value of the analysis file. IMO `checkRecompilations` should only read the last persisted analysis file and make sure that the invariants are met. I'm happy to leave that for another pull request.